### PR TITLE
server: use selector to schedule region peer

### PR DESCRIPTION
**Pull request by huachaohuang**
We will replace balancer with scheduler in the future, but we should
extract some common code from balancer first.

### CODE REVIEW

**Comment by huachaohuang** [comment]
PTAL @siddontang @overvenus

File: server/scheduler.go [hunk 1]
```diff
@@ -35,3 +35,27 @@ func scheduleLeader(cluster *clusterInfo, s Selector) (*regionInfo, *storeInfo,
 
 	return region, source, target
 }
+
+func scheduleStorage(cluster *clusterInfo, s Selector) (*regionInfo, *storeInfo, *storeInfo) {
```
Comment (siddontang) [inline server/scheduler.go]: s/scheduleStorage/scheduleStore/

can we add an explanation for the three results?

**Comment by siddontang** [comment]
Rest LGTM

File: server/scheduler.go (hunk 1 above)
Comment (huachaohuang) [inline server/scheduler.go]: The function name indicates what kind of resource it schedules, and this function schedules for storage.

**Pull request merged**